### PR TITLE
Optimize linkis-cli #4637

### PR DESCRIPTION
**Pull request by Fuu3214**
### What is the purpose of the change

Optimized linkis-cli

### Related issues/PRs

Related issues: #4636

### Brief change log

- Optimized and simplified linkis-cli code

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [x] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by casionone** [comment]
```
```

File: linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/operator/ujes/UJESClientFactory.java [hunk 1]
```diff
@@ -145,8 +141,8 @@ public static DWSClientConfig generateDWSClientConfigForBML(
                       .retryEnabled(false)
                       .readTimeout(context.getReadTimeoutMills())
                       .setAuthenticationStrategy(authenticationStrategy)
-                      .setAuthTokenKey(authKey)
-                      .setAuthTokenValue(authValue)))
+                      .setAuthTokenKey("BML-AUTH")
```
Comment (peacewong) [inline linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/operator/ujes/UJESClientFactory.java:144]: need auth key and token

File: linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/operator/ujes/UJESConstants.java [hunk 2]
```diff
@@ -40,4 +40,7 @@ public class UJESConstants {
   public static final Integer IDX_FOR_LOG_TYPE_ALL = 3; // 0: Error 1: WARN 2:INFO 3: ALL
 
   public static final int DEFAULT_PAGE_SIZE = 500;
+
+  public static final String DEFAULT_SPARK_ENGINE = "spark-2.4.3";
+  public static final String DEFAULT_HIVE_ENGINE = "hive-1.2.1";
```
Comment (peacewong) [inline linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/operator/ujes/UJESConstants.java:45]: is this necessary？

File: linkis-computation-governance/linkis-client/linkis-cli/src/test/java/org/apache/linkis/cli/application/LinkisClientApplicationTest.java [hunk 3]
```diff
@@ -39,19 +35,27 @@ public void before() {
...
-          //        "-varMap", "name=\"databases\""
-
+          "--gatewayUrl",
+          "http://127.0.0.1:9001",
+          "--authStg",
+          "token",
+          "--authKey",
+          "Validation-Code",
+          "--authVal",
+          "BML-AUTH",
```
Comment (peacewong) [inline linkis-computation-governance/linkis-client/linkis-cli/src/test/java/org/apache/linkis/cli/application/LinkisClientApplicationTest.java:45]: need to use conf key

File: linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/interactor/job/common/LogData.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+public class LogData {
+  private String user;
+  private String jobId;
+  private String execId;
+  private LinkedBlockingDeque<String> logBuffer = new LinkedBlockingDeque();
+  private String logPath; // remote path for job log
+  private Integer nextLogLineIdx;
+  private Boolean hasNextLogLine;
+  private volatile Boolean logFin = false;
```
Comment (peacewong) [inline linkis-computation-governance/linkis-client/linkis-cli/src/main/java/org/apache/linkis/cli/application/interactor/job/common/LogData.java:34]: This attribute needs to be commented to explain what it means

**Review by peacewong — APPROVED** [review]
LGTM.

**Comment by aiceflower** [comment]
Compilation failed, test failed

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**